File: Domain/Notice.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace MessingSystem.Domain
{
    public class Notice
    {
        [Key]
        public int Id { get; set; }
        public string Title { get; set; }
        public string Message { get; set; }
        public DateTime Date { get; set; }
        public int CreatedBy { get; set; }
        public bool IsVisible { get; set; }
    }
}

File: wwwroot/js/models/notice.js
var Notice = function (data) {
    var self = this;
    if (data) {
        ko.mapping.fromJS(data, {}, self);
    }
    else {
        self.date = ko.observable(moment(new Date()).format('yyyy-MM-DD'));
        self.title = ko.observable();
        self.message = ko.observable();
        self.isVisible = ko.observable(false);
    }
}